pkg/client/dtclient/settings_client.go

```diff
@@ -231,6 +231,32 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 		return DynatraceEntity{}, fmt.Errorf("unable to generate external id: %w", err)
 	}
 
+	// If the server contains two configs, one with the origin-object-id and a second config with the externalID,
+	// it is not possible to update the setting using the externalId and origin-object-id on the same POST request,
+	// as two settings objects can be the target of the change. In this case, we remove the origin-object-id
+	// and only update the object using the externalId.
+	settings, err := d.listSettings(ctx, obj.SchemaId, ListSettingsOptions{
+		Filter: func(object DownloadSettingsObject) bool {
+			return object.ObjectId == obj.OriginObjectId || object.ExternalId == externalID
+		},
+	})
+	if err != nil {
+		return DynatraceEntity{}, err
+	}
+	if len(settings) == 2 {
+		var exIdSetting, ooIdSetting string
+		if settings[0].ExternalId == externalID {
+			exIdSetting = settings[0].ObjectId
+			ooIdSetting = settings[1].ObjectId
+		} else {
+			exIdSetting = settings[1].ObjectId
+			ooIdSetting = settings[0].ObjectId
+		}
+
+		log.WithCtxFields(ctx).Warn("Found two configs, one with the originObjectId (%q), and one with the externalId (%q). Updating the one with the externalId.", ooIdSetting, exIdSetting)
```

Comment:
I don't think it's fully clear to users what the "externalId" even is. 
I will rebase this PR and try to rephrase this directly,